common/src/main/java/com/redhat/cloud/notifications/models/validation/HttpsNonInternalUrlValidator.java

```diff
@@ -0,0 +1,49 @@
+package com.redhat.cloud.notifications.models.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.UnknownHostException;
+
+
+public class HttpsNonInternalUrlValidator implements ConstraintValidator<HttpsNonInternalUrl, String> {
+
+    private static final String SCHEME_HTTPS = "https";
+
+    @Override
+    public void initialize(HttpsNonInternalUrl constraintAnnotation) {
+        ConstraintValidator.super.initialize(constraintAnnotation);
+    }
+
+    /**
+     * Checks that the URL is well-formed, that the URI syntax is correct, and finally that the host doesn't point to
+     * an internal IP.
+     * @param rawUrl the URL to check.
+     * @param constraintValidatorContext the validation context.
+     * @return true if the provided URL is a well-formed, HTTPS URL which points to an external IP.
+     */
+    @Override
+    public boolean isValid(String rawUrl, ConstraintValidatorContext constraintValidatorContext) {
```

Comment:
I don't think it's good to keep it all on the same validation, it's probably worth to split into more validators, but it's good now!

The reason is to provide a better error message:
- Invalid url
- Unable to resolve host
- Internal addresses are not allowed

Reply:
Can you believe that I had developed three annotations, had them tested, and then be like "nah, probably I'm going to get told to put them all in one"? :rofl: 

All right, I'll work on that!